test-runner/src/Editor.js

```diff
@@ -242,31 +277,35 @@ const setText = (codeMirrorRef, content) => {
   codeMirrorRef.current.dispatch({ effects: scrollEffect });
 };
 
-const setErrors = (codeMirrorRef, errors) => {
-  if (!codeMirrorRef.current || !errors) return;
-  console.log(errors);
+function combineOutputsType(consoleOutputs) {
+  return consoleOutputs.some((output) => output.type === "error")
+    ? "error"
+    : "log";
+}
 
-  const existingErrorState = codeMirrorRef.current.state.field(errorState);
+function groupByLineNumber(consoleOutputs) {
+  return consoleOutputs.reduce((acc, output) => {
+    acc[output.lineNumber] = [...(acc[output.lineNumber] || []), output];
+    return acc;
+  }, {});
+}
 
-  const existingErrors = [];
-  for (let iter = existingErrorState.iter(); iter.value !== null; iter.next()) {
-    existingErrors.push(iter.value);
-  }
+const setConsoleOutputs = (codeMirrorRef, consoleOutputs) => {
+  if (!codeMirrorRef.current || !consoleOutputs) return;
+
+  const outputsByLine = groupByLineNumber(consoleOutputs);
+
+  const effects = Object.entries(outputsByLine).map(([lineNumber, outputs]) => {
+    return consoleEffect.of({
+      from: codeMirrorRef.current.state.doc.line(lineNumber).from,
+      to: codeMirrorRef.current.state.doc.line(lineNumber).to,
+      value: outputs.map((output) => output.message),
+      type: combineOutputsType(outputs),
+    });
+  });
 
   codeMirrorRef.current.dispatch({
-    effects: errors
-      .map((error) => {
-        return errorEffect.of({
-          from: codeMirrorRef.current.state.doc.line(error.line).from,
-          to: codeMirrorRef.current.state.doc.line(error.line).to,
-          error: error.message,
-        });
-      })
-      .filter((error) => {
```

Comment:
No longer need this filter because of: https://github.com/mdbenjam/testing-library-visualizer/pull/39/files#r833385524